src/go/cmd/module.go

```diff
@@ -279,42 +180,33 @@ func parseModFile(dir string) (*modfile.File, error) {
 	return modfile.Parse(p, content, nil)
 }
 
+// recursiveFindTypeDef finds a type definition in a collection of packages.
+//
+//   - typeName is the unqualified name of the type e.g. "RetryOptions"
+//   - source is the package containing the type alias
+//   - packages is the collection of Pkgs to search. Its keys are import paths e.g.
+//     "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
 func recursiveFindTypeDef(typeName string, source *Pkg, packages map[string]*Pkg) (typeDef, bool) {
 	def, ok := source.types[typeName]
 	if ok {
 		return def, true
 	}
-
-	// this is a type alias.  recursively find its typeDef
-	alias, ok := source.typeAliases[typeName]
-	if ok {
-		// alias == github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container.DeleteOptions
-		split := strings.LastIndex(alias, ".")
-		if split < 0 {
-			return typeDef{}, false
-		}
-
-		pkgName := alias[:split]   // github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container
-		typeName = alias[split+1:] // DeleteOptions
-		split = strings.LastIndex(pkgName, "/")
-		if split < 0 {
-			return typeDef{}, false
-		}
-
-		pkgName = pkgName[split+1:] // container
-		source = nil
-		for _, pkg := range packages {
-			if pkg.p.Name == pkgName {
-				source = pkg
+	// source doesn't define typeName; it must export typeName via an alias.
+	// Recurse into the package from which source imports typeName.
+	for _, a := range source.TypeAliases {
+		if a.Name == typeName {
+			// a.QualifiedName == github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container.DeleteOptions
+			dot := strings.LastIndex(a.QualifiedName, ".")
+			if dot < 0 {
```

Comment:
We should also error out if `.` is the last character found in a qualifiedname.. Though maybe you have protection for that elsewhere. However if we ever did hit it, then you'd hit an index out of range on line 204

Reply:
`0 < dot <= len(a.QualifiedName)-2` should always be true and is in fact true for the entire SDK today. I'm uncertain it's worth reworking this function to return an error when this condition doesn't hold because that would entail dropping the `, ok` idiom in favor of returning an error and we'd need a sentinel error for "definition not found". Seems like too much ceremony to catch an improbable bug. I suppose we could panic instead; that condition really should always be true.

Reply:
Agreed that a `panic` for something that should never happen is preferred.